### DISCUSSION TITLE
DataAnnotations / partial class definitionen

### DISCUSSION THREAD

**Discussion by hbrischke** [stephanstapel/ZUGFeRD-csharp#425] [Ideas]
Hallo,
ich würde es gut finden, wenn die Klassen als partial deklariert würden, damit bestünde die Möglichkeit, DataAnnotations hinzuzufügen ohne die original-Dateien ändern zu müssen.
Oder, habe ich etwas übersehen, dass dies bereits auch jetzt schon möglich ist?
Beispiel:
[MetadataType(typeof(ContactMetadata))]
public partial class Contact
{ }
public sealed class ContactMetadata
{

    [Display(Name = "Kontakt-Name")]
    public string Name { get; set; }

    [Display(Name = "Organisationseinheit")]
    public string OrgUnit { get; set; }

    [Display(Name = "Email")]
    public string EmailAddress { get; set; }

    [Display(Name = "Telefonnummer")]
    public string PhoneNo { get; set; }

    [Display(Name = "Faxnummer")]
    public string FaxNo { get; set; }
}

**Comment by stephanstapel** [comment]
@hbrischke : wofür würdest Du das einsetzen wollen?

**Reply by hbrischke** [reply]
... ich arbeite gerade an einem Viewer, WPF. Da würden die Felder dann automatisch (nach einmaliger Definition der Metadaten) 'lesbarer' beschriftet sein und man könnte das auch in verschiedenen Sprachen anzeigen lassen.
![image](https://github.com/user-attachments/assets/f4c67f73-dd3f-4c19-88f3-fbae7ce90591)

**Reply by stephanstapel** [reply]
hmm. Wäre es nicht sauberer, das in eigene Klassen zu mappen. Mit AutoMapper ist das ziemlich simpel erreichbar und ist auch nicht mehr Tipparbeit als wenn Du auf partial-Klassen aufsetzt.

**Comment by hbrischke** [comment]
das kann sein, ich kenne AutoMapper nicht. Schaue ich mir mal an.
In meinen Augen wäre die partial Deklaration auch eine saubere Lösung, die auf den Basis-Mechanismen aufsetzt. ... es führen (wie so oft) viele Wege zum Ziel.
